packages/@o3r/rules-engine/src/engine/operator/operators/array-based.operators.ts

```diff
@@ -119,8 +119,14 @@ export const allLower: Operator<number[], number | string> = {
 export const allMatch: Operator<string[], string> = {
   name: 'allMatch',
   evaluator: (array, inputString) => {
-    const regExp = new RegExp(inputString);
-    return array.every((elementValue) => regExp.test(elementValue));
+    if (inputString.startsWith('/')) {
+      const finalSlash = inputString.lastIndexOf('/');
+      const regexpPattern = inputString.slice(1, finalSlash);
+      const regexpFlags = inputString.slice(finalSlash + 1);
+      const regexp = new RegExp(regexpPattern, regexpFlags);
+      return array.every((elementValue) => regexp.test(elementValue));
+    }
+    return array.every((elementValue) => RegExp(inputString).test(elementValue));
```

Comment:
Reduce the cost if you instantiate the regExp only once:
```suggestion
    const regExp = new RegExp(inputString);
    return array.every((elementValue) => regExp.test(elementValue));
```